x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToDatePeriodTests.java

```diff
@@ -14,45 +14,60 @@
 import org.elasticsearch.xpack.esql.expression.function.FunctionName;
 import org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier;
 import org.elasticsearch.xpack.esql.type.EsqlDataTypeConverter;
 
+import java.time.Period;
+import java.time.temporal.TemporalAmount;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.function.Supplier;
 
+import static org.elasticsearch.xpack.esql.EsqlTestUtils.randomLiteral;
+import static org.elasticsearch.xpack.esql.core.type.DataType.DATE_PERIOD;
+import static org.elasticsearch.xpack.esql.core.type.DataType.KEYWORD;
+import static org.elasticsearch.xpack.esql.core.type.DataType.TEXT;
+import static org.elasticsearch.xpack.esql.type.EsqlDataTypeConverter.DATE_PERIODS;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.matchesPattern;
+
 @FunctionName("to_dateperiod")
-public class ToDatePeriodTests extends AbstractFunctionTestCase {
+public class ToDatePeriodTests extends AbstractScalarFunctionTestCase {
     public ToDatePeriodTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testCaseSupplier) {
         this.testCase = testCaseSupplier.get();
     }
 
     @ParametersFactory
     public static Iterable<Object[]> parameters() {
-        // TODO multivalue fields
-        String read = "Attribute[channel=0]";
         List<TestCaseSupplier> suppliers = new ArrayList<>();
 
-        TestCaseSupplier.unary(
-            suppliers,
-            read,
-            validDatePeriod(DataType.DATE_PERIOD),
-            DataType.DATE_PERIOD,
-            bytesRef -> EsqlDataTypeConverter.parseTemporalAmount(((BytesRef) bytesRef).utf8ToString(), DataType.DATE_PERIOD),
-            List.of()
-        );
-
-        for (DataType inputType : AbstractConvertFunction.STRING_TYPES) {
-            TestCaseSupplier.unary(
-                suppliers,
-                read,
-                validDatePeriod(inputType),
-                DataType.DATE_PERIOD,
-                bytesRef -> EsqlDataTypeConverter.parseTemporalAmount(((BytesRef) bytesRef).utf8ToString(), DataType.DATE_PERIOD),
-                List.of()
+        suppliers.add(new TestCaseSupplier(List.of(DATE_PERIOD), () -> {
+            Period field = (Period) randomLiteral(DATE_PERIOD).value();
+            return new TestCaseSupplier.TestCase(
+                List.of(new TestCaseSupplier.TypedData(field, DATE_PERIOD, "field").forceLiteral()),
+                matchesPattern("LiteralsEvaluator.*"),
+                DATE_PERIOD,
+                equalTo(field)
             );
+        }));
+
+        for (EsqlDataTypeConverter.INTERVALS interval : DATE_PERIODS) {
+            for (DataType inputType : List.of(KEYWORD, TEXT)) {
+                suppliers.add(new TestCaseSupplier(List.of(inputType), () -> {
+                    BytesRef field = new BytesRef(
+                        randomIntBetween(0, 36500000) + " ".repeat(randomIntBetween(1, 10)) + interval.toString()
```

Comment:
nit: negative periods (leading minus, maybe with/without a whitespace in between?) + random leading and trailing whitespaces would be nice!

Reply:
The valid negative intervals(e.g. -3 days, -4 hours) are added in `ToDatePeriodTests` and `ToTimeDurationTests`, the invalid negative intervals(e.g. - 3 days, - 4 hours) are added in `LogicalPlanOptimizerTests`.